shared/test-helpers/assertThrow.js

```diff
@@ -1,26 +1,37 @@
-function assertError(error, s, message) {
-  assert.isAbove(error.message.search(s), -1, `${message}\n\t(error: ${error})`);
+const THROW_ERROR_PREFIX = 'VM Exception while processing transaction:'
+
+function assertError(error, expectedErrorCode) {
+  assert(error.message.search(expectedErrorCode) > -1, `Expected error code "${expectedErrorCode}" but failed with "${error}" instead.`)
 }
 
-async function assertThrows(block, message, errorCode) {
+async function assertThrows(blockOrPromise, expectedErrorCode) {
   try {
-    await block()
-  } catch (e) {
-    return assertError(e, errorCode, message)
+    (typeof blockOrPromise === 'function') ? await blockOrPromise() : await blockOrPromise
+  } catch (error) {
+    assertError(error, expectedErrorCode)
+    return error
   }
-  assert.fail('should have thrown before')
+  assert.fail(`Expected "${expectedErrorCode}" but it did not fail`)
```

Comment:
It could be useful to have the expected revert reason as part of the failure message if it didn't revert